src/components/map/LayersService.js

```diff
@@ -678,7 +678,17 @@ goog.require('ga_urlutils_service');
 
         // Returns promise of layer with its features when layers are added.
         this.getLayerPromise = function(bodId) {
-          return geojsonPromises[bodId];
+          var p, config = this.getLayer(bodId);
+          if (config && /geojson/.test(config.type)) {
+            p = geojsonPromises[bodId];
+            if (!p) {
+              // Fill the geojsonPromises array
+              // TODO: Not nice to create a layer and not using it.
+              this.getOlLayerById(bodId);
```

Comment:
we could also a new method `getGeojsonLayerPromise` and we could avoid the layer creation.
What do you think?

Reply:
We could have a try in another PR. I don't want to creates more bugs in this one.